Source/OxyPlot/Annotations/ImageAnnotation.cs

```diff
@@ -310,5 +298,16 @@ protected ScreenVector GetVector(PlotLength x, PlotLength y, PlotModel model)
 
             return new ScreenVector(xd, yd);
         }
+
+        /// <inheritdoc/>
+        public override OxyRect GetClippingRect()
+        {
+            if (this.X.Unit == PlotLengthUnit.Data || this.Y.Unit == PlotLengthUnit.Data)
+            {
+                return base.GetClippingRect();
+            }
+
+            return new OxyRect(0, 0, double.PositiveInfinity, double.PositiveInfinity);
```

Comment:
> Should we introduce a standard Everything OxyRect (like DataPoint.Invalidate) to which we can compare in the code that sets the clipping rectangle to check it isn't an 'Everything' OxyRect?

Good idea!

> Also, the ImageSharp render context won't cope with out-of-bounds clipping rects

Actually I guess it doesn't have to when we implement your first suggestion.

> Thinking about it further, that would be useful if GetClippingRect where moved to PlotElement, but might also create confusion if it can't be used to splat a rectangle over the whole screen when passed as an argument to a drawing method (which I happen to do in some code I have, though I can't remember why). Perhaps a better option would be to use PlotModel.PlotBounds.

I am not quite sure I get your point here? We do clip to plot bounds anyway [here](https://github.com/Jonarw/oxyplot/blob/41b6c24ebff1bb646bdca62ad6fb3629461b72be/Source/OxyPlot/PlotModel/PlotModel.Rendering.cs#L49), so out-of-bounds clipping rects will be trimmed down [here](https://github.com/Jonarw/oxyplot/blob/41b6c24ebff1bb646bdca62ad6fb3629461b72be/Source/OxyPlot/Rendering/RenderContext/ClippingRenderContext.cs#L49).

Reply:
That last comment was a bit jumbled, I shouldn't worry about; trimming should cover everything smartly.